x-pack/metricbeat/module/sql/query/query_integration_test.go

```diff
@@ -27,20 +27,38 @@ import (
 )
 
 type testFetchConfig struct {
-	Driver string
-	Query  string
-	Host   string
+	Driver            string
+	Query             string
+	Host              string
+	SQLResponseFormat string
 
 	Assertion func(t *testing.T, event beat.Event)
 }
 
 func TestMySQL(t *testing.T) {
 	service := compose.EnsureUp(t, "mysql")
 	config := testFetchConfig{
-		Driver:    "mysql",
-		Query:     "select table_schema, table_name, engine, table_rows from information_schema.tables where table_rows > 0;",
-		Host:      mysql.GetMySQLEnvDSN(service.Host()),
-		Assertion: assertFieldNotContains("service.address", ":test@"),
+		Driver:            "mysql",
+		Query:             "select table_schema, table_name, engine, table_rows from information_schema.tables where table_rows > 0;",
+		Host:              mysql.GetMySQLEnvDSN(service.Host()),
+		SQLResponseFormat: table,
+		Assertion:         assertFieldNotContains("service.address", ":test@"),
+	}
+
+	t.Run("fetch", func(t *testing.T) {
+		testFetch(t, config)
+	})
+
+	t.Run("data", func(t *testing.T) {
+		testData(t, config, "")
```

Comment:
These "data" tests generate the `data.json` example files. The string value is a path for this file, if no path is set the default path `_meta/data.json` is used. We shouldn't use repeated values because then files would be overwritten when regenerating them.
If you want to add example data files (what would be great to see how this change works!), please ensure that this string value is not repeated.
```suggestion
		testData(t, config, "./_meta/data_mysql_tables.json")
```

These files are regenerated by running the integration tests with the `-data` flag, e.g:
```
go test -tags integration ./x-pack/metricbeat/module/sql/query/ -data
```